core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1032,10 +1106,23 @@ public long createFile(AlluxioURI path, CreateFileOptions options)
       BlockInfoException, IOException, FileDoesNotExistException {
     Metrics.CREATE_FILES_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
-        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
-      mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE);
+        MasterAuditContext auditContext =
+             createAuditContext("createFile", path, null, null)) {
+      if (options.isRecursive()) {
+        auditContext.setSrcInode(inodePath.getLastExistingInode());
+      } else {
+        auditContext.setSrcInode(inodePath.getParentInodeOrNull());
```

Comment:
optional: create the audit context with `inodePath.getParentInodeOrNull()`, then overwrite it if `options.isRecursive`.